app/views/contexts/calculated_summary_context.py

```diff
@@ -126,8 +134,8 @@ def _build_calculated_summary_section(
         return {"id": section_id, "groups": [{"id": group["id"], "blocks": blocks}]}
 
     def _remove_unwanted_questions_answers(
-        self, block, answer_ids_to_keep, current_location
-    ) -> str:
+        self, block: Block, answer_ids_to_keep: list, current_location: Location
+    ) -> Any:  # As QuestionnaireSchema.get_mutable_deepcopy returns Any
```

Comment:
I am not sure if we can do that...
As:
self._placeholder_renderer.render returns QuestionnaireSchema.get_mutable_deepcopy(dict_to_render) which returns data which is Any.
Not sure if we can change it to dict

Reply:
The var that is assigned to `QuestionnaireSchema.get_mutable_deepcopy(transformed_block)` needs to be explicitly typed. Anything that returns `Any` can be explicitly typed with type ignores.

Reply:
For example, this can be fixed by:

```py
    def _remove_unwanted_questions_answers(
        self,
        block: Mapping[str, Any],
        answer_ids_to_keep: Iterable[str],
        current_location: Location,
    ) -> dict:
        """
        Evaluates questions in a block and removes any questions not containing a relevant answer
        """
        transformed_block = transform_variants(
            block,
            self._schema,
            self._metadata,
            self._response_metadata,
            self._answer_store,
            self._list_store,
            current_location=current_location,
        )
        transformed_block = deepcopy(transformed_block)
        mutable_transformed_block: dict = QuestionnaireSchema.get_mutable_deepcopy(transformed_block)
        block_question = mutable_transformed_block["question"]

        matching_answers = []
        for answer_id in answer_ids_to_keep:
            matching_answers.extend(self._schema.get_answers_by_answer_id(answer_id))

        questions_to_keep = [
            self._schema.parent_id_map[answer["id"]] for answer in matching_answers
        ]

        if block_question["id"] in questions_to_keep:
            answers_to_keep = [
                answer
                for answer in block_question["answers"]
                if answer["id"] in answer_ids_to_keep
            ]
            block_question["answers"] = answers_to_keep

        return mutable_transformed_block
```

Reply:
👍